LocalGround/localground
# Map Image Processing Working

## Changes
* /api/0/map-images/ endpoint now returns uuid
* /api/0/map-images/<id> endpoint returns uuid, and also allows user to set the status via PUT / PATCH. This is useful in the event that the user wants to re-process the map.
* Any reference to Attachment & Snippet has been deprecated from /localground/apps/lib/image_processing/processor.py script.

Image processing is working again

## Testing Instructions
### New installations needed
```shell
$ sudo apt-get -y install openjdk-7-jre
$ sudo pip install numpy==1.8
```
### Manual testing
Generate a print, print it out, draw on it with colored markers, scan it, and upload it using either the API endpoint or using http://localhost:7777/upload/map-images/. Then...
```shell
$ vagrant ssh
$ cd /localground/apps/lib/jobs/
$ python process-maps.py 
```
When the script completes, go to http://localhost:7777/api/0/map-images/, and check that the "overlay_path" link downloads a processed map. Note that the map UI isn't implemented yet, but this will allow Scot to get started on celery testing.
